index.js

```diff
@@ -132,7 +133,11 @@ module.exports = (opts = {}) => {
 
     selectIpfsApiOpts: state => state.ipfs.apiOpts,
 
-    selectIpfsApiAddress: state => `/ip4/${state.ipfs.apiOpts.host}/tcp/${state.ipfs.apiOpts.port}`,
+    selectIpfsApiAddress: state => {
+      const ipv = state.ipfs.apiOpts.host.split('.').length === 4 ? 'ip4' : 'ip6'
+
+      return `/${ipv}/${state.ipfs.apiOpts.host}/tcp/${state.ipfs.apiOpts.port}`
```

Comment:
if the user sets a domain name as the host then the multiaddr should start with `/dnsaddr` not `/ip*`

It'd be best to use  http://multiformats.github.io/js-multiaddr/ and pick out the info from there. If the api for that is awkward than we can improve it.